aea/aea_builder.py

```diff
@@ -618,14 +607,33 @@ def _check_configuration_not_already_added(self, configuration) -> None:
             configuration.component_id
             in self._package_dependency_manager.all_dependencies
         ):
-            raise ValueError(
-                "Component {} of type {} already added.".format(
+            raise AEAException(
+                "Component '{}' of type '{}' already added.".format(
                     configuration.public_id, configuration.component_type
                 )
             )
 
-    def _check_package_dependencies(self, configuration):
-        self._package_dependency_manager.check_package_dependencies(configuration)
+    def _check_package_dependencies(
```

Comment:
why private?

Reply:
because it is not supposed to be used from the outside. It does not return a `bool` but raises an exception if there's something wrong.

However, I agree it could be useful to have a public method to check if a package can be added. We should discuss how to expose such method (does it return a boolean? Or raises an exception?)

Reply:
A public method should probably return a `bool`. Can be done later though (YAGNI)